pkgs/applications/video/mediathekview/default.nix

```diff
@@ -8,35 +8,61 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-vr0yqKVRodtXalHEIsm5gdEp9wPU9U5nnYhMk7IiPF4=";
   };
 
-
-  nativeBuildInputs = [ makeWrapper zip ];
+  nativeBuildInputs = [ copyDesktopItems makeWrapper zip ];
 
   installPhase =
   let
-    libraryPath = lib.strings.makeLibraryPath [ libglvnd libnotify ];
+    libraryPath = lib.makeLibraryPath ([ libglvnd libnotify ]
+      ++ lib.optional stdenv.isLinux xorg.libXxf86vm);
+    # add JVM args from
+    # https://github.com/mediathekview/MediathekView/blob/9105485f50ec10d863727b4817c8c4ffcbb02643/pom.xml#L136-L139
+    jvmArgList = [
+      "-XX:+UseShenandoahGC"
+      "-XX:ShenandoahGCHeuristics=compact"
+      "-XX:MaxRAMPercentage=50.0"
+      "-XX:+UseStringDeduplication"
+      "--add-opens=java.desktop/sun.awt.X11=ALL-UNNAMED"
+      "-Dfile.encoding=UTF-8"
+      "-DexternalUpdateCheck" # disable checking for updates
+    ];
+    jvmArgs = lib.strings.concatStringsSep " " jvmArgList;
   in
   ''
     runHook preInstall
 
-    mkdir -p $out/{bin,lib}
+    mkdir -p $out/{bin,lib} $out/share/pixmaps
 
     install -m644 MediathekView.jar $out/lib
+    install -m644 MediathekView.svg $out/share/pixmaps
 
     makeWrapper ${jre}/bin/java $out/bin/mediathek \
-      --add-flags "-jar $out/lib/MediathekView.jar" \
+      --add-flags "${jvmArgs} -jar $out/lib/MediathekView.jar" \
       --suffix LD_LIBRARY_PATH : "${libraryPath}"
 
     makeWrapper ${jre}/bin/java $out/bin/MediathekView \
       --add-flags "-jar $out/lib/MediathekView.jar" \
```

Comment:
This line needs to be removed. I get an error when runnung MediathekView:

```
./results/mediathekview/bin/MediathekView
Picked up _JAVA_OPTIONS: -Dsun.java2d.uiScale=2
Unknown options: '-XX:+UseShenandoahGC', '-XX:ShenandoahGCHeuristics=compact', '-XX:MaxRAMPercentage=50.0', '-XX:+UseStringDeduplication', '--add-opens=java.desktop/sun.awt.X11=ALL-UNNAMED', '-Dfile.encoding=UTF-8', '-DexternalUpdateCheck', '-jar'
```

Reply:
Besides that LGTM, but I can't test the desktop items part, as I'm using a tiling wm.

Reply:
> This line needs to be removed. I get an error when runnung MediathekView:

Nice catch! Fixed. BTW, do you know why there are two scripts that start the app? I just tested `mediathek`...

> Besides that LGTM, but I can't test the desktop items part, as I'm using a tiling wm.

I use xmonad myself. And rofi. It shows up there.

Reply:
> > This line needs to be removed. I get an error when runnung MediathekView:
> 
> Nice catch! Fixed. BTW, do you know why there are two scripts that start the app? I just tested `mediathek`...

Those where the scripts in the binary release version for Linux in previous versions. I haven't looked at that archive for some time, but it looks like things have changed in https://download.mediathekview.de/stabil/MediathekView-14.0.0-linux.tar.gz

There are now two launcher scripts (like on windows), ```MediathekView``` and ```MediathekView_Portable```, seemingly generated by install4j.

I guess we could either generate our own scripts with those names and VM options, try to patch the upstream scripts, or (finally) build the package ourselves.

The easiest option, generating our own wrappers like before, would be ok for now I think. Maybe we don't even need the portable version, though it seems to be included in the Deb package.

We should add it to the release note though that the older wrappers have been removed.